### PR TITLE
[obligations] Minor refactoring

### PR DESCRIPTION
**Pull request by ejgallego**
Minors bits here and there, mostly self-explanatory.

### CODE REVIEW

File: vernac/declare.ml [hunk 1]
```diff
@@ -1215,9 +1215,10 @@ let get_hide_obligations =
     ~key:["Hide"; "Obligations"]
     ~value:false
 
-let declare_obligation prg obl body ty uctx =
+let declare_obligation prg obl ~uctx ~types ~body =
+  let univs = UState.univ_entry ~poly:prg.prg_poly uctx in
   let body = prg.prg_reduce body in
-  let ty = Option.map prg.prg_reduce ty in
+  let ty = Option.map prg.prg_reduce types in
```
Comment (SkySkimmer) [inline vernac/declare.ml:1221]: TBH I prefer `typ` to `types` as there's only 1
Also why don't we shadow here?

File: vernac/obligations.ml [hunk 2]
```diff
@@ -254,27 +252,31 @@ and auto_solve_obligations n ?oblset tac : progress =
...
-                                ( Printer.pr_constr_env env sigma x.obl_type
-                                ++ str "." ++ fnl () ) ) )
-                   | Some _ -> ())
+    Array.iteri
+      (fun i x ->
+         match x.obl_body with
+         | None ->
+           if !showed > 0 then (
+             decr showed;
+             show_single_obligation i n obls x)
```
Comment (SkySkimmer) [inline vernac/obligations.ml:278]:
```suggestion
           if !showed > 0 then begin
             decr showed;
             show_single_obligation i n obls x
           end
```
slightly prettier imo

**Review by SkySkimmer — APPROVED** [review]

File: vernac/declare.ml (hunk 1 above)
Comment (ejgallego) [reply vernac/declare.ml]: Shadowing done, I used `types` as this is what has been used on the rest of `Declare`; maybe #12372 would be a good opportunity to discuss a better naming overall.

File: vernac/obligations.ml (hunk 2 above)
Comment (ejgallego) [reply vernac/obligations.ml]: Done, thanks.

**Comment by ejgallego** [comment]
Failure is spurious, windows worker is full.

**Pull request merged**